### PR TITLE
increase overhead to 25K bytes to avoid spurious test failures

### PR DESCRIPTION
**Pull request by dmitris**
as suggested in #20 

Made the tests run in a loop for 10K times to check for possible memory leaks.

### CODE REVIEW

**Comment by tchap** [comment]
I don't have time to look into this now, but what is being solved by running the test 10K times? Won't it just do nothing since the value is already in the tree?

**Comment by dmitris** [comment]
it inserts the value, then deletes the value (as in the current test) - but does that insert/delete cycle not once but multiple times.  (Note that it is not repeated insert of the same value - that would be useless).  If there is a memory leak, it should be greatly amplified through the repetition.

File: patricia/patricia_dense_test.go [hunk 1]
```diff
@@ -197,15 +199,18 @@ func TestTrie_DeleteLeakageDense(t *testing.T) {
 
 	oldBytes := heapAllocatedBytes()
 
-	for _, v := range data {
-		if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
-			t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
+	// repeat insertion/deletion for 10K times to catch possible memory issues
+	for i := 0; i < 10000; i++ {
+		for _, v := range data {
+			if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
```
Comment (tchap) [inline patricia/patricia_dense_test.go]: @dmitris I am failing to see how this is not just inserting the same value 10000 times (once for every `data` iteration).

Comment (dmitris) [inline patricia/patricia_dense_test.go]: it inserts it, then immediately deletes it before going into the next loop iteration - both "insert" and "delete" for loops are inside of the 10K repetition one.  If there is a memory leak, shouldn't that leak be amplified 10000 times and thus get to the noticeable magnitude beyond the GC "noise"?  Sorry if I'm missing a point here...

```
	for i := 0; i < 10000; i++ {
		for _, v := range data {
			if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
				t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
			}
		}

		for _, v := range data {
			if ok := trie.Delete([]byte(v.key)); ok != v.retVal {
				t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
			}
		}
```

Comment (tchap) [inline patricia/patricia_dense_test.go]: Yeah, I guess. Depends what the memory leak is :-) I will check this out when I have more time. Expect me to get back to this aroud 2042 :D

**Pull request merged**